PR: full implementation of standard signature syntax in container signatures

jcharlet opened the pull request:
@nishihatapalmer
> These classes can parse PRONOM and container signature syntax, producing an Abstract Syntax Tree (AST). A compiler for the AST produces ByteSequence, SubSequence and SideFragment objects that can be used in DROID to search for those signatures.

solves https://github.com/digital-preservation/droid/issues/237

jcharlet commented:
@nishihatapalmer the demo for rebasing:
Your commits:
![Screenshot from 2019-11-07 10-54-43](https://user-images.githubusercontent.com/1750370/68384407-77130200-014f-11ea-80bb-9885a42cab13.png)

My simple rebase command: worked right away without any conflict:
from feature branch
`git rebase master`
equivalent to `git pull --rebase master`
![Screenshot from 2019-11-07 10-55-58](https://user-images.githubusercontent.com/1750370/68384404-767a6b80-014f-11ea-8654-d66c6e2e5925.png)

The result:
![Screenshot from 2019-11-07 10-55-00](https://user-images.githubusercontent.com/1750370/68384406-767a6b80-014f-11ea-9d3c-40410ee0f21b.png)

I think it's better for maintenance because when we merge back onto master, we don't have crazy branches all other the place. you can also merge with rebase on github and have a clean history without even needing a merge commit.

jcharlet commented:
if you want, I'll let you do it to practice :) . 

I'll start fixing the checkstyle errors for travis and add documentation, to better understand your work .

nishihatapalmer commented:
OK, I think that worked.  I checked out, rebased, then pushed up.  

Hopefully nothing is messed up!

jcharlet commented:
![Screenshot from 2019-11-07 11-48-50](https://user-images.githubusercontent.com/1750370/68386743-982a2180-0154-11ea-959d-4ad60f684b70.png)
ok brilliant, pushed that on this repo's branch, thanks!

jcharlet commented:
@nishihatapalmer I added a couple commits to fix checkstyle issues as you'll see below. Would you like to have a look to see if it's helpful or not? 
I assumed it would be simpler to simply commit all our work to the same upstream branch and avoid  to have multiple merge commits on the same piece of work. You could check the commits and add comments on things I got wrong, so that I fix them on a later commit. Or maybe you have another way of working together on a branch, and I'm happy to follow whatever you advise.

nishihatapalmer commented:
Sounds good to me.  Shall I add you as a collaborator on my fork, or do you
have another method?

On Thu, 7 Nov 2019 at 16:19, Jeremie Charlet <notifications@github.com>
wrote:

> @nishihatapalmer <https://github.com/nishihatapalmer> I added a couple
> commits to fix checkstyle issues as you'll see below. Would you like to
> have a look to see if it's helpful or not?
> I assumed it would be simpler to simply commit all our work to the same
> upstream branch and avoid to have multiple merge commits on the same piece
> of work. You could check the commits and add comments on things I got
> wrong, so that I fix them on a later commit. Or maybe you have another way
> of working together on a branch, and I'm happy to follow whatever you
> advise.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/digital-preservation/droid/pull/305?email_source=notifications&email_token=ABBY4JHFREQCDPXB7WUUWNTQSQ5ZVA5CNFSM4JKE6MBKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEDM6VJY#issuecomment-551152295>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABBY4JHKQAYNCOCUA2VTLYTQSQ5ZVANCNFSM4JKE6MBA>
> .
>

jcharlet commented:
@nishihatapalmer I was thinking of pushing our commits to droidsyntaxparser branch on the official https://github.com/digital-preservation/droid (not mine or your fork, to avoid having to add user rights on any repo). this is why I created this new PR just here. 
>  Draft	jcharlet wants to merge 42 commits into master from droidsyntaxparser

but if you prefer to add me as a collaborator on your fork's branch and work on it, go ahead, I'll push my stuff there!

jcharlet commented:
ups

nishihatapalmer commented:
**Status update.**

I've now got a reasonably well tested parser and compiler.  It passes all the tests for syntax and building the right sorts of objects from the abstract syntax tree.  There are no more obvious tests to write - all major components are now tested.

The next step is to wire it into the DROID signature XML so it can be compiled by DROID.

nishihatapalmer commented:
**Status update**

Latest code pushed can handle direct compilation of existing container signatures.

Edit a container signature file.   Put the value of the Sequence element under a ByteSequence as a "Sequence" attribute of a ByteSequence.  This will cause DROID to compile the ByteSequence itself from the value of the sequence attribute.

Note, if there are SubSeqMinOffset and SubSeqMaxOffsets defined, you need to put those in front of the sequence element as a bounded range.  e.g. if the ByteSequence has SubSeqMinOffset = 10, SubSeqMaxOffset = 1024, an  Sequence element value is "01 02 03 04", then you have the following Sequence attribute:

`Sequence="{10,1024} 01 02 03 04"`

For example:

```
<ContainerSignature Id="1020" ContainerType="OLE2">
      <Description>Microsoft Word 97 OLE2</Description>
      <Files>
        <File>
          <Path>WordDocument</Path>
        </File>
        <File>
          <Path>CompObj</Path>
          <BinarySignatures>
            <InternalSignatureCollection>
              <InternalSignature ID="300">
                <ByteSequence*Reference="BOFoffset" Sequence="{40-1024} 10 00 00 00 'Word.Document.8' 00">
                  <SubSequence Position="1" SubSeqMinOffset="40"
                    SubSeqMaxOffset="1024">
                    <Sequence>10 00 00 00 'Word.Document.8' 00</Sequence>
                  </SubSequence>
                </ByteSequence>
              </InternalSignature>
            </InternalSignatureCollection>
          </BinarySignatures>
        </File>
      </Files>
    </ContainerSignature>
```

nishihatapalmer commented:
At this point, you could also remove all the old XML it would still work - but if you keep the old XML, DROID warns that it's clearing pre-defined objects in order to perform compilation.  It still works.

nishihatapalmer commented:
** ByteSequenceSerializer**

There is now a ByteSequenceSerializer which can output XML from ByteSequence objects, or transform them into PRONOM syntax in a variety of flavours (e.g. binary, container) with compile targets of droid or pronom (pronom tends to compile shorter anchoring sequences).

*SigTool**
I've created a simple command line tool which can transform signature syntax into XML directly using the serializer.

It can also parse signature files, and either:

1.  output all the signatures with compiled PRONOM syntax sequences in a tab-delimitted format (for both binary or container signature files).
2.  transform the ByteSequence objects in the XML so they strip out everything except a compiled PRONOM expression.   

For example, compiling the most recent V95 binary signature file, outputting new XML using container signature syntax, we get the following:

```
...
        <InternalSignature ID="70" Specificity="Specific">
            <ByteSequence Reference="BOFoffset" Sequence="FF D8 FF E8 00 ' SPIFF' 00 01 00 [00:04] {11} [00:05] {9} FF E8  "/>
        </InternalSignature>
        <InternalSignature ID="72" Specificity="Specific">
            <ByteSequence Reference="BOFoffset" Sequence="09 00 04 00 {2} 10 00  "/>
        </InternalSignature>
...
```

nishihatapalmer commented:
Next steps are to test DROID using a fully "compacted" sig file, which will then only use internally compiled ByteSequence objects, and compare to running with the original file.  The results should be identical, if they're not there's a problem somewhere.

nishihatapalmer commented:
Update:

I can replace all container signatures with the new "Sequence" attribute containing PRONOM syntax, and it gives identical results to the existing container signature file.  This gives some confidence that things are on the right track.

I get very different results when I replace existing binary signatures with their pronom syntax counterparts.  Matching is all over the place.  

Strange that the more complex syntax works without a hitch, but the older binary syntax is failing somewhere.  Maybe its something to do with how the objects are being built or prepared.  This path is different for binary and container signatures.

nishihatapalmer commented:
This is looking like a serialization error in ByteSequenceSerializer for alternative sequences.  They can end up nested in the current implemention, which is illegal.

nishihatapalmer commented:
Think I've found a problem - EOFoffset byte sequences aren't compiling properly.

Should have specific tests for this adding.

nishihatapalmer commented:
Fix was that the EOFoffset was never being set - everything was being compiled as BOFoffset.

Setting the anchor correctly when compiling fixes all issues.  DROID now matches files identically in tests using either old style signature files, or the new ones with PRONOM syntax compiled directly in DROID.

jcharlet commented:
as discussed with Matt on the phone (thaaaaaaaannnkkks!)

Matt: test new complex signature
Jeremie: 
- fix checkstyle issues and make it run on CI
- run this end to end test on skeleton suite, maybe include it in regression test cases
- build jar of sig tool CLI with maven, and provide bat sh scripts to run it

jcharlet commented:
ok @nishihatapalmer brilliant, and it packages locally without any issue.
then I'm left with running the end test, if you don't do it by the time my train arrives at Kew =)) ;) !...

a small detail: instead of duplicating the sigtool readme in droid-tools and droid-binary, shouldn't we simply state in droid-binary that a full description of the commands and results is available in droid-binary ?

jcharlet commented:
and actually leave the file in .md format in droid-binary (to let know people that it's a markdown document, and so that we can open directly in the right format in our IDE). I'll do it if you find it's a good idea.

jcharlet commented:
I'm going to rebase the master into your branch (and my work on bundling with the jre), you'll have to pull changes before making any new update. I'll let you know here when it's done

jcharlet commented:
probably have to modify the win related assembly file too

jcharlet commented:
ok started updating the regression test using skeleton suite. Am going to do that work in another branch and will merge back on this one to verify. 
Also just hit the weird graphical issue caused by a lib update that we discussed on the phone Matt, so will merge that before as well.

nishihatapalmer commented:
Hi Jeremie,

great - it will be a good test that all is working properly (using an
original and transformed signature file I assume!).

I'm now reasonably happy with what's there.  The sigtool can now be used to
test container signatures as well as binary signatures.  I'll have a go at
running a "full syntax" container signature tomorrow, and I'm sure I'll
find a few more tweaks and fixes, but I think it's basically feature
complete.

Please do fix the UI issues (and merge in the PRs I've given for filter
enums and the NPE on the export button).  I'm currently having to manually
put those fixes in just so I can actually use DROID for any real testing!

cheers,

Matt.

On Thu, 14 Nov 2019 at 12:35, Jeremie Charlet <notifications@github.com>
wrote:

> ok started updating the regression test using skeleton suite. Am going to
> do that work in another branch and will merge back on this one to verify.
> Also just hit the weird graphical issue caused by a lib update that we
> discussed on the phone Matt, so will merge that before as well.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/digital-preservation/droid/pull/305?email_source=notifications&email_token=ABBY4JDOOURK4E5FUDYRTYTQTVA2HA5CNFSM4JKE6MBKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEEBV4FA#issuecomment-553868820>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABBY4JDMR4D4WLTUWIMOKQ3QTVA2HANCNFSM4JKE6MBA>
> .
>

jcharlet commented:
Brilliant!

I'll definitely merge all your PR and will work on the UI issue first next
week. And then I will merge onto your branch and merge it. It's just that
I'm stuck on the UI to check on the skeleton suite to validate this work
:).

On Thu, 14 Nov 2019 at 21:42, nishihatapalmer <notifications@github.com>
wrote:

> Hi Jeremie,
>
> great - it will be a good test that all is working properly (using an
> original and transformed signature file I assume!).
>
> I'm now reasonably happy with what's there. The sigtool can now be used to
> test container signatures as well as binary signatures. I'll have a go at
> running a "full syntax" container signature tomorrow, and I'm sure I'll
> find a few more tweaks and fixes, but I think it's basically feature
> complete.
>
> Please do fix the UI issues (and merge in the PRs I've given for filter
> enums and the NPE on the export button). I'm currently having to manually
> put those fixes in just so I can actually use DROID for any real testing!
>
> cheers,
>
> Matt.
>
> On Thu, 14 Nov 2019 at 12:35, Jeremie Charlet <notifications@github.com>
> wrote:
>
> > ok started updating the regression test using skeleton suite. Am going to
> > do that work in another branch and will merge back on this one to verify.
> > Also just hit the weird graphical issue caused by a lib update that we
> > discussed on the phone Matt, so will merge that before as well.
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/digital-preservation/droid/pull/305?email_source=notifications&email_token=ABBY4JDOOURK4E5FUDYRTYTQTVA2HA5CNFSM4JKE6MBKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEEBV4FA#issuecomment-553868820
> >,
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/ABBY4JDMR4D4WLTUWIMOKQ3QTVA2HANCNFSM4JKE6MBA
> >
> > .
> >
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/digital-preservation/droid/pull/305?email_source=notifications&email_token=AANLKYTRGH75UIFMGWDS6VDQTXA47A5CNFSM4JKE6MBKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEEDMW4Q#issuecomment-554093426>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AANLKYSAYD3J6ZV7QSYONHDQTXA47ANCNFSM4JKE6MBA>
> .
>

nishihatapalmer commented:
You could avoid duplicating the sigtool readme in both sigtool and binary.  It needs to be in binary so that users get it. 

I just changed the file extension to .txt since a lot of people won't know what a markdown file is (and it may not open in a text editor by default in Windows).

nishihatapalmer commented:
The other thing which is missing from all of this is an explanation of what the new capabilities in DROID actually are.  sigtool is a nice tool, but it wasn't actually the point of doing this work in the first place!

We need to explain somewhere that container signatures can now be specified in full PRONOM syntax, by placing the syntax in a ByteSequence sequence attribute:

`<ByteSequence reference="BOFoffset" sequence="01 02 (03|04) ?? [20:40] 'some text' FF" />`

nishihatapalmer commented:
In uk.gov.nationalarchives.droid.core.signature.compiler there is another markdown file which explains the valid PRONOM / container syntax.

That would also be a useful resource to put in binary.

jcharlet commented:
good points, I'll do those 3.

nishihatapalmer commented:
One more comment (probably!).  You can also put the ByteSequence Sequence attribute into a binary signature file signature - it's not just limited to container signatures.

I guess most people won't bother, in the sense that they get their binary signatures from the PRONOM service, which will only publish traditional signatures.

It does mean that anyone who wants to experiment with a new binary signature can add it quickly into a binary signature file to try it with DROID.   Even if when submitted to TNA, it eventually gets PRONOM compiled back into XML again for formal release.

jcharlet commented:
Hi @nishihatapalmer ,

I've updated the documentation above if you want to have a look. Build now passing too.

> One more comment (probably!). You can also put the ByteSequence Sequence attribute into a binary signature file signature - it's not just limited to container signatures.

I've been spending some time to understand how to do that and I must say I feel confused. 
I guess I have to:
- add a String sequence attribute to InternalSignature
- add in InternalSignature.prepareForUse a call to a method that converts string to list of bytesequences.  
But how to do that conversion?
Been looking at how you convert binary to container signature in sigtool and the method `use ByteSequenceCompiler.compile(sequence, binaryExpression, CompileType.DROID)` but it converts the bytes into a string.

nishihatapalmer commented:
You don't have to modify any more code  - it's already there and working.

You just put a Sequence attribute on a ByteSequence element in the XML (instead of having loads of XML sub-elements like SubSequence and LeftFragment, etc).

You can just write 

`<ByteSequence Reference="BOFoffset" Sequence="010203040506" />`

nishihatapalmer commented:
The sigtool can convert between binary and container format (it defaults to container format, so it will put strings in rather than bytes).  Just specify --binary (or -b) on the command line to force binary-compatible output.

I should say however, that even the binary signatures can be compiled using the full container syntax.  You don't have to put binary syntax into the binary file and container syntax into the container file.  As far as the compiler and parser is concerned, it's all valid DROID syntax.

If you were submitting a binary signature back to TNA, you might want to put it in binary syntax (since PRONOM doesn't understand the newer container syntax).  DROID itself is fine with either.

nishihatapalmer commented:
The documentation is a bit confused.  You talk about getting binary signature files from PRONOM, then talk about putting ByteSequences into container files.  You're really talking about three different things:

1. What kinds of signature file are there?
2.  Where do we get updated signature files from?
3.  What are the new syntax capabilities?

1.  Binary and container signatures are supported by DROID 6.0 onwards.

2.  Download location:  both binary and container files are downloaded from TNA, although binary files use a SOAP protocol, whereas container files are just HTTP downloads (which is much simpler - I wish binary files did this too).

3.  New capabilities.  This is what you're really describing - using the Sequence attribute on a ByteSequence element.

I would keep the original paragraph you cut in your commit (which talks about what signature files are and where they are available).

Then I would add  under Signatures Managment:

Since version 6.5, DROID can compile signatures itself, without needing a full XML specification in either the binary or container signature files.  Inside signature files, the actual sequences to match are specified under <ByteSequence> elements.  For example:

```
<ByteSequence Reference="BOFoffset>
  <SubSequence SubSeqMinOffset="10" SubSeqMaxOffset="1024">
    <Sequence>0102030405[00:30]</Sequence>
  <SubSequence/>
  <SubSequence>
    <Sequence>01 02 03</Sequence>
</SubSequence/>
```

DROID can now simply take a full PRONOM signature inside a new Sequence attribute on the ByteSequence.  For example, the signature above could be written as:

`<ByteSequence Reference="BOFoffset" Sequence="{10-1024} 01 02 03 04 05 [00:30] * 01 02 03" />`

nishihatapalmer commented:
This doesn't touch on the differences between binary and container syntax.  DROID itself doesn't care, but PRONOM won't understand container syntax.

We should probably link to a description of the syntax somewhere, and the fact that you can use all of the syntax in either binary or container signatures if you like (but PRONOM won't be able to parse them for binary signatures if you want to submit them to TNA).  Of course, nothing stops TNA using sigtool to rewrite a container signatures syntax in binary compatible format.

jcharlet commented:
Aaaaallright sorry @nishihatapalmer for the confusion, some things are indeed a bit clearer, and I better understood your previous comment. Thanks for taking the time to explain!

> We should probably link to a description of the syntax somewhere, 

I mentioned in README.md "PRONOM Syntax also provides details on the regular expression syntax supported by DROID.", which I am going to improve with a relative link on github to the .md file. Isn't it sufficient? 

> and the fact that you can use all of the syntax in either binary or container signatures if you like (but PRONOM won't be able to parse them for binary signatures if you want to submit them to TNA). Of course, nothing stops TNA using sigtool to rewrite a container signatures syntax in binary compatible format.

what about adding the following paragraph, after what you offered to put in `Signatures management` and before the introduction to sigtool `"To work further on signatures, we provide Sigtool [...]"`

_The full syntax can be used in either binary or container signatures. However, if you use the new syntax, to submit them to TNA and get those signatures included in PRONOM registry, you will need to compile those signatures for PRONOM using Sigtool._

nishihatapalmer commented:
I'm going to be pedantic here, but it's kind of important.   

Only binary signatures need to be in binary format.  So if you submit a binary signature to TNA, it should be in binary format.

Container signatures submitted to TNA can - and should - use the full syntax, since PRONOM doesn't compile those, and it allows things which binary signatures don't support.

nishihatapalmer commented:
And if you don't intend to submit the signatures to TNA, you can use the full syntax in the Sequence attribute in either binary or container signatures in DROID.

jcharlet commented:
@nishihatapalmer could you please refine the README? I won't be as clear as you can be, I'd rather let you do it if you have some time, and that will save us from going back and forth here.

nishihatapalmer commented:
Sure, but may be a few days before I'll have time.



On Wed, 27 Nov 2019, 16:16 Jeremie Charlet, <notifications@github.com>
wrote:

> @nishihatapalmer <https://github.com/nishihatapalmer> could you please
> refine the README? I won't be as clear as you can be, I'd rather let you do
> it if you have some time, and that will save us from going back and forth
> here.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/digital-preservation/droid/pull/305?email_source=notifications&email_token=ABBY4JHT7JCGKKO3KLXVTT3QV2MO7A5CNFSM4JKE6MBKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEFKAORQ#issuecomment-559155014>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABBY4JCXJNJBHMR26EETXLLQV2MO7ANCNFSM4JKE6MBA>
> .
>

nishihatapalmer commented:
OK, I've posted some updates to the README.md file in my droidsyntaxparser
branch.

I'm tempted to simplify it a bit - remove the "Types" and "Syntax" sections
(or considerably shorten them, or move them into another markdown file).
The real new capabilities are "sigtool" and "Simpler signature XML".

I kind of feel that putting in a quick education on what kinds of
signatures exist in the middle of a high level overview of what's in the
DROID repo is the wrong place for this - but telling users about sigtool
and simpler signature XML is OK, as they're big and important changes.

On Wed, 27 Nov 2019 at 21:02, Matt Palmer <mattpalms@gmail.com> wrote:

> Sure, but may be a few days before I'll have time.
>
>
>
> On Wed, 27 Nov 2019, 16:16 Jeremie Charlet, <notifications@github.com>
> wrote:
>
>> @nishihatapalmer <https://github.com/nishihatapalmer> could you please
>> refine the README? I won't be as clear as you can be, I'd rather let you do
>> it if you have some time, and that will save us from going back and forth
>> here.
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/digital-preservation/droid/pull/305?email_source=notifications&email_token=ABBY4JHT7JCGKKO3KLXVTT3QV2MO7A5CNFSM4JKE6MBKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEFKAORQ#issuecomment-559155014>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ABBY4JCXJNJBHMR26EETXLLQV2MO7ANCNFSM4JKE6MBA>
>> .
>>
>

jcharlet commented:
thanks for those changes!

what about 
- renaming PRONOM Syntax to signature syntax.
- moving types and syntax section in that signature syntax readme.
- add a paragraph below `Since version 6.5, DROID adds some new capabilities to support developing and testing signatures.` : 
> Signature Syntax provides details on the types of signatures and regular expression syntax supported by DROID.

nishihatapalmer commented:
Yup, those sound perfect.

On Thu, 28 Nov 2019, 14:53 Jeremie Charlet, <notifications@github.com>
wrote:

> thanks for those changes!
>
> what about
>
>    - renaming PRONOM Syntax to signature syntax.
>    - moving types and syntax section in that signature syntax readme.
>    - add a paragraph below Since version 6.5, DROID adds some new
>    capabilities to support developing and testing signatures. :
>
> Signature Syntax provides details on the types of signatures and regular
> expression syntax supported by DROID.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/digital-preservation/droid/pull/305?email_source=notifications&email_token=ABBY4JBJO6FSFBY3UE7PV3DQV7LPJA5CNFSM4JKE6MBKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEFM2J2I#issuecomment-559523049>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABBY4JCUSYJPP2UQXZVSLNLQV7LPJANCNFSM4JKE6MBA>
> .
>

nishihatapalmer commented:
As long as we retain a link to the signature syntax MD from the readme


On Thu, 28 Nov 2019, 14:59 Matt Palmer, <mattpalms@gmail.com> wrote:

> Yup, those sound perfect.
>
> On Thu, 28 Nov 2019, 14:53 Jeremie Charlet, <notifications@github.com>
> wrote:
>
>> thanks for those changes!
>>
>> what about
>>
>>    - renaming PRONOM Syntax to signature syntax.
>>    - moving types and syntax section in that signature syntax readme.
>>    - add a paragraph below Since version 6.5, DROID adds some new
>>    capabilities to support developing and testing signatures. :
>>
>> Signature Syntax provides details on the types of signatures and regular
>> expression syntax supported by DROID.
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/digital-preservation/droid/pull/305?email_source=notifications&email_token=ABBY4JBJO6FSFBY3UE7PV3DQV7LPJA5CNFSM4JKE6MBKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEFM2J2I#issuecomment-559523049>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ABBY4JCUSYJPP2UQXZVSLNLQV7LPJANCNFSM4JKE6MBA>
>> .
>>
>

nishihatapalmer commented:
I'm not sure the current link is actually working come to think of it...

On Thu, 28 Nov 2019, 15:03 Matt Palmer, <mattpalms@gmail.com> wrote:

> As long as we retain a link to the signature syntax MD from the readme
>
>
> On Thu, 28 Nov 2019, 14:59 Matt Palmer, <mattpalms@gmail.com> wrote:
>
>> Yup, those sound perfect.
>>
>> On Thu, 28 Nov 2019, 14:53 Jeremie Charlet, <notifications@github.com>
>> wrote:
>>
>>> thanks for those changes!
>>>
>>> what about
>>>
>>>    - renaming PRONOM Syntax to signature syntax.
>>>    - moving types and syntax section in that signature syntax readme.
>>>    - add a paragraph below Since version 6.5, DROID adds some new
>>>    capabilities to support developing and testing signatures. :
>>>
>>> Signature Syntax provides details on the types of signatures and regular
>>> expression syntax supported by DROID.
>>>
>>> —
>>> You are receiving this because you were mentioned.
>>> Reply to this email directly, view it on GitHub
>>> <https://github.com/digital-preservation/droid/pull/305?email_source=notifications&email_token=ABBY4JBJO6FSFBY3UE7PV3DQV7LPJA5CNFSM4JKE6MBKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEFM2J2I#issuecomment-559523049>,
>>> or unsubscribe
>>> <https://github.com/notifications/unsubscribe-auth/ABBY4JCUSYJPP2UQXZVSLNLQV7LPJANCNFSM4JKE6MBA>
>>> .
>>>
>>

jcharlet commented:
fixed https://github.com/digital-preservation/droid/blob/droidsyntaxparser/README.md

> I'm not sure the current link is actually working come to think of it...
> […](#)
> On Thu, 28 Nov 2019, 15:03 Matt Palmer, ***@***.***> wrote: As long as we retain a link to the signature syntax MD from the readme On Thu, 28 Nov 2019, 14:59 Matt Palmer, ***@***.***> wrote: > Yup, those sound perfect. > > On Thu, 28 Nov 2019, 14:53 Jeremie Charlet, ***@***.***> > wrote: > >> thanks for those changes! >> >> what about >> >> - renaming PRONOM Syntax to signature syntax. >> - moving types and syntax section in that signature syntax readme. >> - add a paragraph below Since version 6.5, DROID adds some new >> capabilities to support developing and testing signatures. : >> >> Signature Syntax provides details on the types of signatures and regular >> expression syntax supported by DROID. >> >> — >> You are receiving this because you were mentioned. >> Reply to this email directly, view it on GitHub >> <#305?email_source=notifications&email_token=ABBY4JBJO6FSFBY3UE7PV3DQV7LPJA5CNFSM4JKE6MBKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEFM2J2I#issuecomment-559523049>, >> or unsubscribe >> <https://github.com/notifications/unsubscribe-auth/ABBY4JCUSYJPP2UQXZVSLNLQV7LPJANCNFSM4JKE6MBA> >> . >> >

fixed

jcharlet commented:
done!

nishihatapalmer commented:
Looks great

By the way, why does the standard non jre version of DROID have a Unix
suffix on the filename?  It should be completely platform independent.

On Thu, 28 Nov 2019, 15:21 Jeremie Charlet, <notifications@github.com>
wrote:

> done!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/digital-preservation/droid/pull/305?email_source=notifications&email_token=ABBY4JEQ6473OAYMOSOXESTQV7OXDA5CNFSM4JKE6MBKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEFM4QIQ#issuecomment-559532066>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABBY4JHEAC2VLIZCEUDJOGTQV7OXDANCNFSM4JKE6MBA>
> .
>

nishihatapalmer commented:
One more comment, the signature syntax file now talks about the sigtool
"below".  It's not below now as that's still in the original readme.  Maybe
change to a link back to the readme or the using sigtool documentation?



On Thu, 28 Nov 2019, 15:48 Matt Palmer, <mattpalms@gmail.com> wrote:

> Looks great
>
> By the way, why does the standard non jre version of DROID have a Unix
> suffix on the filename?  It should be completely platform independent.
>
> On Thu, 28 Nov 2019, 15:21 Jeremie Charlet, <notifications@github.com>
> wrote:
>
>> done!
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/digital-preservation/droid/pull/305?email_source=notifications&email_token=ABBY4JEQ6473OAYMOSOXESTQV7OXDA5CNFSM4JKE6MBKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEFM4QIQ#issuecomment-559532066>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ABBY4JHEAC2VLIZCEUDJOGTQV7OXDANCNFSM4JKE6MBA>
>> .
>>
>

jcharlet commented:
> Looks great By the way, why does the standard non jre version of DROID have a Unix suffix on the filename? It should be completely platform independent.
> […](#)
> On Thu, 28 Nov 2019, 15:21 Jeremie Charlet, ***@***.***> wrote: done! — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub <#305?email_source=notifications&email_token=ABBY4JEQ6473OAYMOSOXESTQV7OXDA5CNFSM4JKE6MBKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEFM4QIQ#issuecomment-559532066>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ABBY4JHEAC2VLIZCEUDJOGTQV7OXDANCNFSM4JKE6MBA> .

mmh it used to because we had a windows version with jre and a unix version without jre. 
But shouldn't be the case anymore, since we recently put back the windows stuff in the bundle without jre and thus removed the unix suffix.
just checked on master and droidsyntaxparser on the core repo here, and it looks good to me https://github.com/digital-preservation/droid/blob/master/droid-binary/assembly.xml

maybe I missed something, where did you see that?

jcharlet commented:
> One more comment, the signature syntax file now talks about the sigtool "below". It's not below now as that's still in the original readme. Maybe change to a link back to the readme or the using sigtool documentation?

sorry for that.. fixed
